### PR TITLE
Reduce the styling gap between BE and FE on block editor. [TEC-4250]

### PR DESCRIPTION
**Pull request by EugeneKyale**
Update the backend block editor UI for a single event to look similar to its frontend version in order to create a true WYSIWYG experience when using the block editor.

Ticket: https://theeventscalendar.atlassian.net/browse/TEC-4250

https://www.loom.com/share/851a132c94f044e4961eeea2bb6e8556

### CODE REVIEW

File: src/Tribe/Views/V2/Hooks.php [hunk 1]
```diff
@@ -68,6 +69,35 @@ protected function add_actions() {
...
+		if ( empty( $post ) || ! $post instanceof WP_Post || ! tribe_is_event( $post ) ) {
+			return;
+		}
+
+		// Append the customizer styles to the single block stylesheet
+		add_filter( 'tribe_customizer_inline_stylesheets', function( $sheets ) {
+			$sheets[] = 'tribe-admin-v2-single-blocks';
+
+			return $sheets;
+		} );
```
Comment (bordoni) [inline src/Tribe/Views/V2/Hooks.php:97]:
```suggestion
		add_filter( 'tribe_customizer_inline_stylesheets', static function( $sheets ) {
			$sheets[] = 'tribe-admin-v2-single-blocks';

			return $sheets;
		} );
```

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Views/V2/Assets.php [hunk 2]
```diff
@@ -634,4 +650,19 @@ public function should_enqueue_single_event_block_editor_styles() {
...
+	 * Load assets on the add or edit pages of the block editor only.
+	 *
+	 * @since  TBD
+	 *
+	 * @return bool
+	 */
+	public function should_enqueue_admin() {
+		if ( is_admin() && get_current_screen()->is_block_editor ) {
+			return true;
+		}
```
Comment (bordoni) [inline src/Tribe/Views/V2/Assets.php:664]: Split these into two separate inverted conditionals returning false and return true on the end.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Views/V2/Assets.php (hunk 2 above)
Comment (EugeneKyale) [reply src/Tribe/Views/V2/Assets.php]: @bordoni done.

File: src/Tribe/Views/V2/Assets.php [hunk 3]
```diff
@@ -634,4 +650,23 @@ public function should_enqueue_single_event_block_editor_styles() {
...
+	 * @return bool
+	 */
+	public function should_enqueue_admin() {
+		if ( ! is_admin() ) {
+			return false;
+		}
+
+		if ( ! get_current_screen()->is_block_editor ) {
+			return false;
+		}
```
Comment (borkweb) [inline src/Tribe/Views/V2/Assets.php:668]: Let's ensure that the assets only load on our post type pages.

```suggestion
		}
		
		if ( ! tribe( 'admin.helpers' )->is_post_type_screen() ) {
			return false;
		}
```

File: src/Tribe/Views/V2/Hooks.php [hunk 4]
```diff
@@ -68,6 +69,35 @@ protected function add_actions() {
...
+	/**
+	 * Enqueue Customizer styles for the single event block editor screen.
+	 *
+	 * @since TBD
+	 */
+	public function enqueue_customizer_in_block_editor() {
+		// Make sure we're on the block edit screen
+		if ( ! is_admin() || ! get_current_screen()->is_block_editor ) {
+			return;
+		}
```
Comment (borkweb) [inline src/Tribe/Views/V2/Hooks.php:84]: Let's ensure that the assets only load on our post type pages.

```suggestion
		}
		
		if ( ! tribe( 'admin.helpers' )->is_post_type_screen() ) {
			return;
		}
```

**Review by borkweb — CHANGES_REQUESTED** [review]

**Review by borkweb — APPROVED** [review]
🥇

File: src/Tribe/Views/V2/Assets.php [hunk 5]
```diff
@@ -471,6 +471,22 @@ public function register() {
...
 			]
 		);
+
+		tribe_asset(
+			$plugin,
+			'tribe-admin-v2-single-blocks',
+			'tribe-admin-single-blocks.css',
+			[ 
+				'tec-variables-full',
+				'tec-variables-skeleton',
```
Comment (Camwyn) [inline src/Tribe/Views/V2/Assets.php:481]: https://github.com/the-events-calendar/tribe-common/pull/1716 should make this unnecessary

**Review by Camwyn — APPROVED** [review]
Reluctantly approved - my issues are not with your work, but the work this was based on ❤️ as such I'll refrain from commenting on that.

**Review by Camwyn — APPROVED** [review]

**Pull request merged**